### ISSUE TITLE
Load tracking scripts / Change GTM consent params based on user preference

### ISSUE THREAD

**Issue by rcantin-w** [wellcomecollection/wellcomecollection.org#10694]
- Users should not be tracked when first landing on the page. 
- Should they decide not to interact with the cookie banner by dismissing it or by choosing "Essential only", they should not be tracked for the rest of their journey.
- Should they chose "Accept all", scripts should be loaded on the DOM/activated/consent be given to GTM. This should happen seamlessly and without requiring a reload.
 
If there is a change in the consent, all actions done in the background (scripts being added/removed, consent being revoked/accepted) should be seamless and not require a reload.

**Comment by rcantin-w** [comment]
I've had a play with this yesterday and the GTM bit I need to look into more: https://support.google.com/tagmanager/answer/10718549?hl=en

Civic UK does offer a template to control it (https://www.civicuk.com/cookie-control/documentation/gtm-consent-template), but I'm also trying to do it myself for more control. 

The difficulty I think will lie with not affecting our current integration, where consent is given from the get-go, which is the opposite of what we're doing. If any configuration is needed from the GTM platform, it'll affect and probably block tracking until we release the cookie banner to all and the choice to allow tracking is given. So I'll do more thinking on that as well.

**Comment by rcantin-w** [comment]
Now in prod.

So I've configured one new trigger/tag in GTM, both called `consent_btn_test`. They serve as a test for the consent changes and are on the "Cookie preference centre" button, which only shows behind the cookie toggle.

<img src="https://github.com/wellcomecollection/wellcomecollection.org/assets/110461050/e93fdb97-e034-4a29-ac43-8ea88208a5b6" width="200" />


![Image](https://github.com/wellcomecollection/wellcomecollection.org/assets/110461050/43b08feb-f976-4064-85c7-50acb9b39089)

I set the tag to require additional consent from `analytics_storage`, which is why it doesn't fire when clicked when consent is denied.

![Image](https://github.com/wellcomecollection/wellcomecollection.org/assets/110461050/e9ee3ac0-544a-4600-a971-4e8193bdb041)

We'll need to add it to all of our tags when we're ready, _then_ deploy the codebase with the toggle removed. That day will probably have slight wobbles in tracking because of it, but they would be very minor. From then, tags will only fire when people granted analytics permission. 

The Segment script only loads/segment tracking only fires when consent is granted.
